code/game/turfs/open/glass.dm

```diff
@@ -0,0 +1,26 @@
+/turf/open/floor/glass
+	name = "Glass floor"
+	desc = "Dont jump on it, or do, I'm not your mom."
+	icon = 'icons/turf/floors/glass.dmi'
+	icon_state = "glass-0"
+	base_icon_state = "glass"
+	baseturfs = /turf/open/openspace
+	intact = FALSE //this means wires go on top
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_TURF_OPEN, SMOOTH_GROUP_FLOOR_TRANSPARENT_GLASS)
+	canSmoothWith = list(SMOOTH_GROUP_FLOOR_TRANSPARENT_GLASS)
+	footstep = FOOTSTEP_PLATING
+	barefootstep = FOOTSTEP_HARD_BAREFOOT
+	clawfootstep = FOOTSTEP_HARD_CLAW
+	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+
+/turf/open/floor/glass/Initialize()
+	icon_state = "" //Prevent the normal icon from appearing behind the smooth overlays
+	return ..()
```

Comment:
Somewhere here must be `AddElement(/datum/element/turf_transparency, FALSE)`?

Reply:
yes i still need to throw it in